protocols/jvss/jvss.go

```diff
@@ -10,28 +17,554 @@ import (
 // JVSS Protocol Instance structure holding the information for a longterm JVSS
 // signing mechanism
 type JVSSProtocol struct {
-	// The host we are running on
-	Host *sda.Host
-	// The tree we are using
-	Tree *sda.Tree
+	// The TreeNode denating ourself in the tree
+	Node *sda.Node
 	// The EntityList we are using / this is needed to "bypass" the tree
 	// structure for the internals communication, when we set up the shares and
 	// everything. We directly send our share to everyone else directly by using
 	// this entitylist instead of broadcasting into the tree.
-	List *network.EntityList
-	// the token for this protocol instance
-	Token *sda.Token
+	List *sda.EntityList
+	// the index where we are in this entitylist
+	index int
+	// the mapping between TreeNode's peer id in the Tree to index in the entitylist,
+	// since JVSS mostly use the entityList
+	nodeToIndex map[uuid.UUID]int
+	// list of public keys represented in the entityList (needed by poly.Deal)
+	publicList []abstract.Point
+	// keys of the Host set as config.KeyPair
+	key config.KeyPair
+	// The longterm shared private public key pair used in this JVSS.
+	// The idea is that you can keep this protocol instance as a longterm JVSS
+	// using it to distributively sign anything as long as it runs.
+	longterm *poly.SharedSecret
+	// The schnorr struct used to sign / verify using the longterm key
+	schnorr *poly.Schnorr
+	// polyinfo related to how the shares are generated and reconstructed
+	Info poly.Threshold
+	// requests holds all the requests that we asked
+	requests map[int]*RequestBuffer
+	// lastrequestnumber seen or executed
+	lastRequestNo int
+	// more tmp variables  that are needed because of the way we use protocol
+	// isntance. For example, you want to generate a distributed secret, you
+	// need a poly.Receiver struct. You first create that struct in Start()
+	// since you are the leader, and then receive the dealers in Dispatch(),
+	// so you need that intermediate variable.
+	// longterm-receiver of the shares
+	ltReceiver *poly.Receiver
+	// channel through we notify that we have sucessfully computed the longterm
```

Comment:
language: channel through which we .. or: channel that indicates if longterm ...